app/src/main/java/com/twilio/video/app/ui/room/RoomActivity.java

```diff
@@ -286,21 +285,38 @@ protected void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.activity_room);
         ButterKnife.bind(this);
 
+        setupThumbnailRecyclerView();
+
         // Setup toolbar
         setSupportActionBar(toolbar);
 
         // Cache volume control stream
         savedVolumeControlStream = getVolumeControlStream();
 
         // setup participant controller
-        participantController = new ParticipantController(thumbnailLinearLayout, primaryVideoView);
-        participantController.setListener(participantClickListener());
+        primaryParticipantController = new PrimaryParticipantController(primaryVideoView);
 
         // Setup Activity
         statsScheduler = new StatsScheduler();
         obtainVideoConstraints();
     }
 
+    private void setupThumbnailRecyclerView() {
```

Comment:
Yeah maybe I'm missing something here, but it does seem like we could use LiveData here instead of RX.